nemo/collections/tts/models/magpietts.py

```diff
@@ -284,39 +284,39 @@ def audio_to_codes(self, audio, audio_len, audio_type='target'):
             raise ValueError(f"Received audio_type of {audio_type}. Must be `target` or `context`")
 
         self._codec_model.eval()
-        with torch.cuda.amp.autocast(enabled=False):
-            with torch.no_grad():
-                codes, codes_len = self._codec_model.encode(audio=audio, audio_len=audio_len)
-                # Add a timestep to begining and end of codes tensor
-                bos_tensor = torch.full(
-                    (codes.size(0), codes.size(1), 1), audio_bos_id, dtype=codes.dtype, device=codes.device
-                )
-                pad_tensor = torch.full(
-                    (codes.size(0), codes.size(1), 1), 0, dtype=codes.dtype, device=codes.device
-                )  # 0 is the padding token in the audio codebook
-                codes = torch.cat([bos_tensor, codes, pad_tensor], dim=-1)
-                # codes: (B, C, T')
-                # codes_len: (B,)
-                for idx in range(codes.size(0)):
-                    codes[idx, :, codes_len[idx] + 1] = audio_eos_id
-                codes_len = codes_len + 2
-
-                return codes.long(), codes_len.long()
+        with torch.no_grad(), torch.autocast(device_type=str(audio.device), dtype=torch.float32):
+            codes, codes_len = self._codec_model.encode(audio=audio, audio_len=audio_len)
+            # Add a timestep to begining and end of codes tensor
+            bos_tensor = torch.full(
+                (codes.size(0), codes.size(1), 1), audio_bos_id, dtype=codes.dtype, device=codes.device
+            )
+            pad_tensor = torch.full(
+                (codes.size(0), codes.size(1), 1), 0, dtype=codes.dtype, device=codes.device
+            )  # 0 is the padding token in the audio codebook
+            codes = torch.cat([bos_tensor, codes, pad_tensor], dim=-1)
+            # codes: (B, C, T')
+            # codes_len: (B,)
+            for idx in range(codes.size(0)):
+                codes[idx, :, codes_len[idx] + 1] = audio_eos_id
+            codes_len = codes_len + 2
+
+            return codes.long(), codes_len.long()
```

Comment:
This may be not a good place to cast to fp32 if `audio` has already been cast to bf16 before feeding into the function `def audio_to_codes()`. So we should disable the cast much earlier before calling `self.audio_to_codes()`. Wrapp disabling autocast at some places where `audio` was loaded.